### PR TITLE
Add Cube.js MongoDB Database Driver

### PR DESCRIPTION
**Pull request by ChrisLahaye**
**Check List**
- [ ] Tests has been run in packages where changes made if available
- [ ] Linter has been run for changed code
- [ ] Tests for the changes have been added if not covered yet
- [ ] Docs have been added / updated if required

**Issue Reference this PR resolves**

#9015

**Description of Changes Made (if issue reference is not provided)**

Add Cube.js MongoDB Database Driver

### CODE REVIEW

**Comment by igorlukanin** [comment]
Hi @ChrisLahaye 🙌 

Thanks for your contribution! I feel like we'd need integrations tests to make sure this one works.

Also, did you have a chance to publish this as a separate NPM package? Then we can put it under the list of [third-party drivers](https://cube.dev/docs/product/configuration/data-sources#third-party-drivers), as suggested by the [contribution policy](https://github.com/cube-js/cube/blob/master/CONTRIBUTING.md#contributing-database-drivers).

**Comment by KSDaemon** [comment]
Hi @ChrisLahaye ! 🙌

Thank you for your contribution!
However, as @igorlukanin correctly noted - we definitely need at least basic integration tests for this to be landed. In other cases, we can't guarantee it is working or will be working with future updates in the project.

Don't worry if you haven't done it before. It's not so hard! You need to spin up a MongoDB docker container and run a few basic queries against it. Please, have a look at some existing integrations as examples:

* There is a good example of setting up everything in one place - [Vertica driver](https://github.com/cube-js/cube/pull/9081)
* Another example is [MongoBiDriver.test](https://github.com/cube-js/cube/blob/master/packages/cubejs-mongobi-driver/test/MongoBiDriver.test.ts)
* Or of course [PostgresDriver.test](https://github.com/cube-js/cube/blob/master/packages/cubejs-postgres-driver/test/PostgresDriver.test.ts)

TLDR; Version:
1. Add MongoDBRunner to `@cubejs-backend/testing-shared`
2. Create a few basic tests (create table, do some basic queries) along the lines of PostgresDriver.test
3. Add `integration:mongo` script or smth like that to package.json
4. And add it to CI push workflow.

Feel free to ask any help if needed.

**Comment by igorlukanin** [comment]
It is better to pursue native, non-JDBC implementations of data source drivers: https://github.com/cube-js/cube/commit/3f71e56b29fb6c82f388b2ab994493317d288c65
